### ISSUE TITLE
多机多卡训练的时候报错

### ISSUE THREAD

**Issue by badmic** [modelscope/swift#1519]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)
在运行多机多卡的脚本的时候，训练到一半出现报错：
remaining_time': '5m 17s'}
{'loss': 0.42493286, 'acc': 0.87409611, 'grad_norm': 4.97665731, 'learning_rate': 7.4e-06, 'memory(GiB)': 19.76, 'train_speed(iter/s)': 1.160588, 'epoch': 0.3, 'global_step/max_steps': '185/619', 'percentage': '29.89%', 'elapsed_time': '2m 13s', 'remaining_time': '5m 13s'}
{'loss': 0.48568544, 'acc': 0.85100479, 'grad_norm': 5.08678424, 'learning_rate': 7.31e-06, 'memory(GiB)': 19.76, 'train_speed(iter/s)': 1.167997, 'epoch': 0.31, 'global_step/max_steps': '190/619', 'percentage': '30.69%', 'elapsed_time': '2m 16s', 'remaining_time': '5m 8s'}
{'loss': 0.46891456, 'acc': 0.85671806, 'grad_norm': 4.15173273, 'learning_rate': 7.23e-06, 'memory(GiB)': 19.76, 'train_speed(iter/s)': 1.172261, 'epoch': 0.32, 'global_step/max_steps': '195/619', 'percentage': '31.50%', 'elapsed_time': '2m 20s', 'remaining_time': '5m 5s'}
{'loss': 0.49436207, 'acc': 0.85269146, 'grad_norm': 4.11789802, 'learning_rate': 7.14e-06, 'memory(GiB)': 19.76, 'train_speed(iter/s)': 1.179493, 'epoch': 0.32, 'global_step/max_steps': '200/619', 'percentage': '32.31%', 'elapsed_time': '2m 23s', 'remaining_time': '5m 0s'}
Train:  32%|█████████████████████████▏                                                    | 200/619 [02:23<04:35,  1.52it/s]
m1:480098:480434 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480095:480350 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480101:480348 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480097:480457 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480099:480468 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480100:480464 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480096:480460 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480102:480447 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480101:480348 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480100:480464 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480102:480447 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480095:480350 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480098:480434 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480099:480468 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480097:480457 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480096:480460 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480101:480348 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480098:480434 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480097:480457 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480095:480350 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480100:480464 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480099:480468 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480102:480447 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480096:480460 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480101:480348 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480102:480447 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480100:480464 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480095:480350 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480098:480434 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480097:480457 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480096:480460 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480099:480468 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : client reregistration

m1:480101:480348 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480102:480447 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480098:480434 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480099:480468 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480095:480350 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480100:480464 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480096:480460 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480097:480457 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port active

m1:480102:480447 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

m1:480101:480348 [0] transport/net_ib.cc:100 NCCL WARN NET/IB : mlx5_8:1 Got async event : port error

运行的脚本为：
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NNODES=2 \
NODE_RANK=0 \
MASTER_ADDR=127.0.0.1 \
NPROC_PER_NODE=8 \
swift sft \
    --model_type qwen1half-7b-chat \
    --sft_type full \
    --dataset blossom-math-zh \
    --output_dir output \
    --deepspeed default-zero3 

CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
NNODES=2 \
NODE_RANK=1 \
MASTER_ADDR=xxxx.xxxx.xxxx.xxxx \
NPROC_PER_NODE=8 \
swift sft \
    --model_type qwen1half-7b-chat \
    --sft_type full \
    --dataset blossom-math-zh \
    --output_dir output \
    --deepspeed default-zero3 

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by Jintao-Huang** [comment]
是在评估的时候 还是 保存模型的时候报错呀

**Comment by badmic** [comment]
训练的时候出现了通信报错，已解决，是NCCL和cuda的版本不匹配导致的，重新安装环境就行了
